src/keysym.c

```diff
@@ -55,6 +55,31 @@
 #include "keysym.h"
 #include "ks_tables.h"
 
+bool
+xkb_keysym_is_assigned(xkb_keysym_t ks)
+{
+    if (ks > XKB_KEYSYM_MAX)
+        return false;
+    if (XKB_KEYSYM_UNICODE_MIN <= ks && ks <= XKB_KEYSYM_UNICODE_MAX)
+        return true;
+    if (ks < keysym_to_name[0].keysym ||
+        ks > keysym_to_name[ARRAY_SIZE(keysym_to_name) - 1].keysym)
+        return false;
+
+    int32_t lo = 0, hi = ARRAY_SIZE(keysym_to_name) - 1;
+    while (hi >= lo) {
+        int32_t mid = (lo + hi) / 2;
+        if (ks > keysym_to_name[mid].keysym) {
+            lo = mid + 1;
+        } else if (ks < keysym_to_name[mid].keysym) {
+            hi = mid - 1;
+        } else {
+            return true;
+        }
+    }
+    return false;
```

Comment:
I keep looking at this and thinking there's a way to re-use this binary search in `xkb_keysym_get_name()`. Maybe through a combination of:
```c
static ssize_t find_offset(xkb_keysym_t ks); /* returns -1 on error or the offset */

bool xkb_keysym_is_assigned(xkb_keysym_t ks) {
    return find_offset(ks) != -1; 
}

int
xkb_keysym_get_name() {
     ssize_t offset = find_offset(ks);
     if (offset != -1)
         return snprintf(get_name(&keysym_to_name[offset]));
     ...
}
```